lib/AST/Type.cpp

```diff
@@ -1590,6 +1597,11 @@ bool TypeBase::isSpelledLike(Type other) {
   llvm_unreachable("Unknown type kind");
 }
 
+LayoutConstraint TypeBase::getLayoutConstraint() {
+  auto CanTy = getCanonicalType();
+  return CanTy.getLayoutConstraint();
```

Comment:
There's no need to have the CanType method for this, just call getAs instead of dyn_cast